src/internal_events/docker_logs.rs

```diff
@@ -188,3 +187,27 @@ impl<'a> InternalEvent for DockerLogsLoggingDriverUnsupportedError<'a> {
         counter!("logging_driver_errors_total", 1);
     }
 }
+
+#[derive(Debug)]
+pub struct DockerLogsReceivedOldLogError<'a> {
+    pub timestamp_str: &'a str,
+    pub container_id: &'a str,
+}
+
+impl InternalEvent for DockerLogsReceivedOldLogError<'_> {
+    fn emit(self) {
+        error!(
+            message = "Received older log.",
```

Comment:
Regardless of whether we keep as internal event or as a trace! , I think we could improve the message as well. Like adding the context of "older than what?" Something like: "Received log timestamped before container creation." (That is just a guess as to what self.created refers to)

Reply:
> Received log timestamped before container creation.

I think you have missed that this is being emitted in the catch all match and not the one that has the comment about container creation. Easy to do, the code is a little confusing, I'll try to tidy it up.